kpops/pipeline.py

```diff
@@ -85,6 +98,67 @@ def to_yaml(self) -> str:
             self.model_dump(mode="json", by_alias=True, exclude_none=True)["components"]
         )
 
+    def build_execution_graph_from(
+        self,
+        components: list[PipelineComponent],
+        reverse: bool,
+        runner: Callable[[PipelineComponent], Coroutine],
+    ) -> Awaitable:
+        sub_graph_nodes = self.__get_graph_nodes(components)
+
+        async def run_parallel_tasks(tasks):
+            asyncio_tasks = []
+            for coroutine in tasks:
+                asyncio_tasks.append(asyncio.create_task(coroutine))
+            await asyncio.gather(*asyncio_tasks)
+
+        async def run_graph_tasks(pending_tasks: list[Awaitable]):
+            for pending_task in pending_tasks:
+                await pending_task
+
+        sub_graph = self.graph.subgraph(sub_graph_nodes)
+        transformed_graph = sub_graph.copy()
+
+        root_node = "root_node_bfs"
+        transformed_graph.add_node(root_node)
```

Comment:
could you explain this step, please? a comment would be helpful

Reply:
Should I do a comment on the code too ?

Reply:
What I do here is to give the whole pipeline graph a root step. In that way is easier to get all parallel jobs in one run :) because I connect all the nodes without an input edge to this node.

Reply:
yes, please add a comment